jax/_src/dtypes.py

```diff
@@ -420,8 +428,8 @@ def _type_promotion_lattice(jax_numpy_dtype_promotion: str) -> dict[JAXType, lis
       b1: [i_],
       u1: [i2, u2], u2: [i4, u4], u4: [i8, u8], u8: [f_],
       i_: [u1, i1], i1: [i2], i2: [i4], i4: [i8], i8: [f_],
-      f_: [*f1_types, bf, f2, c_],
-      **{t: [] for t in f1_types}, bf: [f4], f2: [f4], f4: [f8, c4], f8: [c8],
+      f_: [*f1_types, f1_e4m3fnuz, f1_e5m2fnuz, bf, f2, c_],
+      **{t: [] for t in f1_types}, f1_e4m3fnuz: [f2], f1_e5m2fnuz: [f2], bf: [f4], f2: [f4], f4: [f8, c4], f8: [c8],
```

Comment:
```suggestion
      f_: [*f1_types, bf, f2, c_],
      **{t: [] for t in f1_types}, bf: [f4], f2: [f4], f4: [f8, c4], f8: [c8],
```
`f1` should not have any implicit promotion path to `f2` for now, and if we make the decision to do so, we should do so for *all` float8 types rather than just these two.

See #16832 and accompanying discussion regarding float8 implicit promotion.